src/router/utils/HelperFunctions.jsx

```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
+import Toast from "../../components/Toast";
+
+export const addToWatchlist = async (video, videoDispatch) => {
+  try {
+    const response = await axios.post(
+      "/api/user/watchlater",
+      { video },
+      {
+        headers: {
+          authorization: localStorage.getItem("userToken"),
```

Comment:
Since you're using axios, instead of passing headers in each request, you can move this work to axios onterceptors. That way you won't require passing headers explicitly in each request. It'll be handled by axios interceptors

Reply:
Thank you, I will check that out.